src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,151 @@
+            var alreadyHandled = new HashSet<TypeSyntax>();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                MakeDeclarationNullable(document, editor, node, alreadyHandled);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private void MakeDeclarationNullable(Document document, SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled)
+        {
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix != null && !alreadyHandled.Contains(declarationTypeToFix))
+            {
+                var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix);
+                editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+                alreadyHandled.Add(declarationTypeToFix);
+            }
+        }
+
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                return null;
+            }
+
+            if (node.IsParentKind(SyntaxKind.ReturnStatement))
+            {
+                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsKind(SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration));
+                if (containingMember == null)
+                {
+                    return null;
+                }
+
+                switch (containingMember.Kind())
+                {
+                    case SyntaxKind.MethodDeclaration:
+                        // string M() { return null; }
+                        return ((MethodDeclarationSyntax)containingMember).ReturnType;
+
+                    case SyntaxKind.PropertyDeclaration:
+                        // string x { get { return null; } }
+                        return ((PropertyDeclarationSyntax)containingMember).Type;
+                }
+            }
+
+
+            // string x = null;
+            if (node.Parent?.Parent?.IsParentKind(SyntaxKind.VariableDeclaration) == true)
+            {
+                var variableDeclaration = (VariableDeclarationSyntax)node.Parent.Parent.Parent;
+                if (variableDeclaration.Variables.Count != 1)
+                {
+                    // string x = null, y = null;
+                    return null;
+                }
+
+                return variableDeclaration.Type;
+            }
+
+            // string x { get; set; } = null;
+            if (node.Parent?.IsParentKind(SyntaxKind.PropertyDeclaration) == true)
+            {
+                var propertyDeclaration = (PropertyDeclarationSyntax)node.Parent.Parent;
+                return propertyDeclaration.Type;
+            }
+
+            // void M(string x = null) { }
+            if (node.Parent?.IsParentKind(SyntaxKind.Parameter) == true)
```

Comment:
no need for ? before IsParentKind, the Is helpers do the right thing on null (they are trying to emulate the 'is' language feature).